### ISSUE TITLE
build Warp/PICSAR in the beamsim/Vagrant container

### ISSUE THREAD

**Issue by bruhwiler** [radiasoft/devops#96]
This is important but is not funded by any active projects, so charge time to Science Support or Tech Support.

Warp can use a new highly-optimized physics kernel called PICSAR. The primary goal is to build it on Cori, but we have a collaborator at the University of Strathclyde who wants to the ability to work with Warp on his Windows computer inside a VM.

The Warp source is here (as we all know):
https://bitbucket.org/berkeleylab/warp

The PICSAR home page is here:
https://picsar.net/

The Bitbucket repository is not yet open (will be soon, I hear), but if you send them your Bitbucket username, then they will give you read access.

**Comment by robnagler** [comment]
I started with this, because it was easiest. With a few modifications, it works in our pyenv setup.

Would you want a separate pyenv-virtualenv for this or do you want this in the normal picsar environment? You would have to select the PXR solver anyway so seems like this is the right way to go.

I would make this a "code" that gets installed in our normal beamsim container. I wouldn't compile it with KNL enabled. You could install it with:

```
curl radia.run | bash -s code picsar
```

**Comment by robnagler** [comment]
One thing is that we would need to upgrade to the latest Warp, which is probably not a problem.

**Comment by bruhwiler** [comment]
Yes, we need a recent version of Warp.
By default, I guess the script should clone the latest versions of both.

In the future, it would be nice to have some control over the code versions -- probably specifying a date to get the most recent versions that are still older than the specified date. This is not a high priority.

**Comment by robnagler** [comment]
When is PICSAR scheduled to be released? I don't particularly want to be adding my credentials to the travis build so would be better when it is FOSS.

**Comment by bruhwiler** [comment]
I don't know.

We should not distribute it before it is released, so I suggest this issue be put on hold until that time.

**Comment by robnagler** [comment]
We are waiting on PICSAR to be FOSS. We shouldn't do anything until then. Closing for now.
